### PR TITLE
Add more info about why lookup is in AwaitingDownload

### PR DESCRIPTION
**Pull request by dapplion**
## Issue Addressed

When debugging lookup stuck incidents in https://github.com/sigp/lighthouse/issues/5833, it's very helpful to know if the specific reason why a lookup is at the `AwaitingDownload` state.

## Proposed Changes

As purely informational, append to `AwaitingDownload` enum variant the last reason why the lookup is at that state.

### CODE REVIEW

File: beacon_node/network/src/sync/block_lookups/single_block_lookup.rs [hunk 1]
```diff
@@ -397,7 +411,7 @@ impl<T: Clone> SingleLookupRequestState<T> {
                     });
                 }
                 self.failed_downloading = self.failed_downloading.saturating_add(1);
-                self.state = State::AwaitingDownload;
+                self.state = State::AwaitingDownload("retry");
```
Comment (realbigsean) [inline beacon_node/network/src/sync/block_lookups/single_block_lookup.rs]: Might be good to differentiate "retry after download failure" from "retry after processing failure"

File: beacon_node/network/src/sync/block_lookups/single_block_lookup.rs [hunk 2]
```diff
@@ -364,18 +369,27 @@ impl<T: Clone> SingleLookupRequestState<T> {
...
             false
         }
     }
 
+    /// Append metadata on why this request is in AwaitingDownload status. Very helpful to debug
+    /// stuck lookups. Not fallible as it's purely informational.
+    pub fn update_awaiting_download_status(&mut self, new_status: &'static str) {
+        match &mut self.state {
+            State::AwaitingDownload(status) => *status = new_status,
+            _ => {}
```
Comment (realbigsean) [inline beacon_node/network/src/sync/block_lookups/single_block_lookup.rs]: looks like a lint is failing here

**Review by realbigsean — CHANGES_REQUESTED** [review]
I like the idea!

File: beacon_node/network/src/sync/block_lookups/single_block_lookup.rs (hunk 1 above)
Comment (dapplion) [reply beacon_node/network/src/sync/block_lookups/single_block_lookup.rs]: Changed to "not started" since it will never remain in that state

**Review by realbigsean — APPROVED** [review]

**Comment by realbigsean** [comment]
@mergify queue

**Comment by michaelsproul** [comment]
needs an update

**Comment by jimmygchen** [comment]
@mergify queue

**Pull request merged**